yt_dlp/extractor/neteasemusic.py

```diff
@@ -311,21 +321,28 @@ class NetEaseMusicSingerIE(NetEaseMusicBaseIE):
             'title': '李昇基 - 이승기',
         },
         'playlist_count': 50,
-        'skip': 'Blocked outside Mainland China',
+    }, {
+        'note': 'Singer with both translated and alias',
+        'url': 'https://music.163.com/#/artist?id=159692',
+        'info_dict': {
+            'id': '159692',
+            'title': '初音ミク - 初音未来;Hatsune Miku',
+        },
+        'playlist_count': 50,
     }]
 
     def _real_extract(self, url):
         singer_id = self._match_id(url)
 
         info = self.query_api(
-            'artist/%s?id=%s' % (singer_id, singer_id),
-            singer_id, 'Downloading singer data')
+            f'artist/{singer_id}?id={singer_id}', singer_id, note='Downloading singer data')
 
-        name = info['artist']['name']
-        if info['artist']['trans']:
-            name = '%s - %s' % (name, info['artist']['trans'])
-        if info['artist']['alias']:
-            name = '%s - %s' % (name, ';'.join(info['artist']['alias']))
+        name_and_aliases = traverse_obj(info, (
+            'artist', ('name', 'trans', ('alias', ...)), {str}, {lambda i: i or None}))
+        if len(name_and_aliases) > 1:
+            name = f'{name_and_aliases[0]} - {";".join(name_and_aliases[1:])}'
+        else:
+            name = traverse_obj(name_and_aliases, 0)
```

Comment:
Making this non-fatal is good, but it seems like the behavior of the code is changing slightly.
Assuming values are found for all keys, the old code would produce:
`name - trans - alias1;alias2;alias3`
whereas the new code would produce:
`name - trans;alias1;alias2;alias3`

Could do it with something like this:
```suggestion
        artist = traverse_obj(info, ('artist', {dict})) or {}
        name = join_nonempty('name', 'trans', delim=' - ', from_dict=artist) or None
        if name:
            name = join_nonempty(name, ';'.join(traverse_obj(artist, ('alias', ...))), delim=' - ')
```

Reply:
Actually, this behavior change was somewhat intended.
Original code only contains cases where only one of [`trans`](https://music.163.com/#/artist?id=124098) or [`alias`](https://music.163.com/#/artist?id=10559) exists. However, taking the example of [Hatsune Miku](https://music.163.com/#/artist?id=159692) (also added in testcase), the website actually displays it as `初音ミク  初音未来;Hatsune Miku` in the title, where `trans` and `alias` are treated equally as alt title parts, so I believe `name - trans;alias` is a more "correct" way to display rather than `name - trans - alias`.
Do you think it is worth making the change or it is better to stick to the original one in favor of consistency?

Reply:
> Do you think it is worth making the change or it is better to stick to the original one in favor of consistency?

Changing it is fine, then. I hadn't grasped that nuance, thanks for the explanation.

This may be simpler, if it works. Try it out:
```suggestion
        name = join_nonempty(
            traverse_obj(info, ('artist', 'name', {str})),
            ';'.join(traverse_obj(info, ('artist', ('trans', ('alias', ...)), {str}))),
            delim=' - ')
```